packages/core/src/Page/index.js

```diff
@@ -468,8 +470,15 @@ class Page {
       ...layoutManager.getLayoutPageNjkAssets(this.layout),
     };
 
+    const pageVueRenderFnsScriptFileName = this.getPageVueRenderFnsScriptFileName();
```

Comment:
how about just the file's basename? (+ some prefix maybe) (then output it right beside the page, instead of everything under `assets`)

also since you can't have two files with the same name in any directory - so no need to ensure no conflicts, which is still possible here (`x/y/abc.js` -> `x-y-abc.js`, `x-y-abc.js` -> `x-y-abc.js`).

Reply:
Ah, yup, thanks for the spot! The solution you suggested is a lot more fool-proof. Will look into it!

Reply:
the `_site` output folder should be polluted, which the user shouldn't be concerned with, not the user's source file directories

Reply:
Ah right my bad! I misinterpreted the comment and was confused for a bit.